File: 백준/Bronze/2501. 약수 구하기/약수 구하기.py
import sys

N, K = list(map(int, sys.stdin.readline().split()))
big = (max(N, K))
small = (min(N, K))
count = 0

for i in range(1, big+1):
    if big % i == 0:
        count += 1
        if count == small:
            print(i)
            break

if small > count:
    print(0)

File: 백준/Bronze/2501. 약수 구하기/약수 구하기.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException{

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int count = 0;

        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());

        int big = Math.max(N,K);
        int small = Math.min(N,K);

        for (int i =1; i <= big; i++){
            if (big % i == 0){
                count += 1;
                if (count == K){
                    bw.write(String.valueOf(i));
                }
            }
        }
        if (count < K){
            bw.write(String.valueOf(0));
        }
        br.close();
        bw.flush();
        bw.close();

    }
}